crates/nu-cli/src/syntax_highlight.rs

```diff
@@ -43,6 +43,8 @@ impl Highlighter for NuHighlighter {
             if shape.0.end <= last_seen_span
                 || last_seen_span < global_span_offset
                 || shape.0.start < global_span_offset
+                || (shape.1 == FlatShape::Operator
+                    && (shape.0.end - 1 <= last_seen_span || shape.0.end - 2 <= last_seen_span))
```

Comment:
Those offsets seem oddly specific. Is there a larger principle that is underlying that? Would adding whitespace in an expression break that?

Reply:
No , as I monitored the code whitespace can't be set that as operator . and operator will be used for conditional block . in condition I added span for lower than 3 to mention exact operator which is `not` and there is only operator which coded in Nushell with manual span . others set span as name span .

Reply:
Just one thing. 
All shapes goes to one vector,  if span matchs to calling name it will set color .
All was doing great. And passed in loop if you see , just we added one option to pass operator too .

Reply:
Any new Updates about this PR ?

Reply:
I have to admit that this area of parsing is not part of my expertise so I would welcome others to chime in.

Reply:
@fdncred @rgwood I believe may they can help us

Reply:
We talked about this PR yesterday and the consensus was this fix may work but may be in the wrong place. 

It seems like we'd want to fix the problem with why the span is wrong to begin with other than patching a fix by manipulating a span. Manipulating spans tends to be brittle so it may fix this problem but break somewhere else or break again due to a future change and we don't want to keep patching spans.

So, If you can help us understand why it's broken in the first place, and help us fix the root cause of this, we'd be very grateful.

Reply:
Thanks for explaining that to me. Let's see what others think.